openstack/networking/v2/extensions/lbaas_v2/pools/requests.go

```diff
@@ -0,0 +1,359 @@
+		Persistence    *SessionPersistence `json:"session_persistence,omitempty"`
+		AdminStateUp   *bool               `json:"admin_state_up,omitempty"`
+	}
+
+	type request struct {
+		Pool pool `json:"pool"`
+	}
+
+	reqBody := request{Pool: pool{
+		Name:           opts.Name,
+		Description:    opts.Description,
+		TenantID:       opts.TenantID,
+		SubnetID:       opts.SubnetID,
+		Protocol:       opts.Protocol,
+		LoadbalancerID: opts.LoadbalancerID,
+		ListenerID:     opts.ListenerID,
+		LBMethod:       opts.LBMethod,
+		AdminStateUp:   opts.AdminStateUp,
+	}}
+
+	if opts.Persistence != nil {
+		reqBody.Pool.Persistence = opts.Persistence
+	}
+
+	var res CreateResult
+	_, res.Err = c.Post(rootURL(c), reqBody, &res.Body, nil)
+	return res
+}
+
+// Get retrieves a particular pool based on its unique ID.
+func Get(c *gophercloud.ServiceClient, id string) GetResult {
+	var res GetResult
+	_, res.Err = c.Get(resourceURL(c, id), &res.Body, nil)
+	return res
+}
+
+// UpdateOpts contains the values used when updating a pool.
+type UpdateOpts struct {
+	// Optional. Name of the pool.
+	Name string
+
+	// Optional. Human-readable description for the pool.
+	Description string
+
+	// The algorithm used to distribute load between the members of the pool. The
+	// current specification supports LBMethodRoundRobin, LBMethodLeastConnections
+	// and LBMethodSourceIp as valid values for this attribute.
+	LBMethod string
+
+	// Optional. The administrative state of the Listener. A valid value is true (UP)
+	// or false (DOWN).
+	AdminStateUp *bool
+}
+
+// Update allows pools to be updated.
+func Update(c *gophercloud.ServiceClient, id string, opts UpdateOpts) UpdateResult {
+	type pool struct {
+		Name         string `json:"name,omitempty"`
+		Description  string `json:"description,omitempty"`
+		LBMethod     string `json:"lb_algorithm,omitempty"`
+		AdminStateUp *bool  `json:"admin_state_up,omitempty"`
+	}
+	type request struct {
+		Pool pool `json:"pool"`
+	}
+
+	reqBody := request{Pool: pool{
+		Name:         opts.Name,
+		Description:  opts.Description,
+		LBMethod:     opts.LBMethod,
+		AdminStateUp: opts.AdminStateUp,
+	}}
+
+	// Send request to API
+	var res UpdateResult
+	_, res.Err = c.Put(resourceURL(c, id), reqBody, &res.Body, &gophercloud.RequestOpts{
+		OkCodes: []int{200},
+	})
+	return res
+}
+
+// Delete will permanently delete a particular pool based on its unique ID.
+func Delete(c *gophercloud.ServiceClient, id string) DeleteResult {
+	var res DeleteResult
+	_, res.Err = c.Delete(resourceURL(c, id), nil)
+	return res
+}
+
+// MemberListOpts allows the filtering and sorting of paginated collections through
+// the API. Filtering is achieved by passing in struct field values that map to
+// the Member attributes you want to see returned. SortKey allows you to
+// sort by a particular Member attribute. SortDir sets the direction, and is
+// either `asc' or `desc'. Marker and Limit are used for pagination.
+type MemberListOpts struct {
```

Comment:
they are included as part of pools since you can only operate on members with respect to a specific pool.